python/cudf/cudf/core/udf/masked_typing.py

```diff
@@ -411,6 +413,33 @@ def generic(self, args, kws):
             return nb_signature(MaskedType(types.int64), args[0])
 
 
+@cuda_decl_registry.register_global(abs)
+class MaskedScalarAbsoluteValue(AbstractTemplate):
+    """
+    Typing for the builtin function abs. Returns the same
+    type as input except for boolean values which are converted
+    to integer.
+
+    This follows the expected result from the builtin abs function
+    which differs from numpy - np.abs returns a bool whereas abs
+    itself performs the cast.
+    """
+
+    def generic(self, args, kws):
+        if isinstance(args[0], MaskedType):
+            if isinstance(args[0].value_type, (StringView, UDFString)):
+                # reject string types
+                return
+            else:
+                return_type = self.context.resolve_function_type(
+                    self.key, (args[0].value_type,), kws
+                ).return_type
```

Comment:
This line uses numba's internals to obtain the result of typing `abs` if the inputs were scalars.

Reply:
I realise this might be opening pandora's box (again) here, but I'll just point out that pandas behaves inconsistently for the `INT_MIN` case depending on where or not you spelt it `s.apply(abs)` or `s.apply(np.abs)`. This is because `np.abs` preserves dtype, and (AFAICT) invokes C-level UB when called on `INT_MIN`. In contrast, `abs` will convert to `PyObject`, perform arbitrary width integer computation and get the right answer, which pandas will massage into an unsigned type.

I am happy to return a column with an unsigned dtype though.

Reply:
(I think we are OK to return an unsigned type in both cases TBH). `abs(INT_MIN)` is UB so we can "do what we like"